master/static/migrations/20240306144927_trial-profiler-metrics-migration.tx.up.sql

```diff
@@ -0,0 +1,65 @@
+/*
+  Migrates existing profiling metrics (in `trial_profiler_metrics`) to `metrics`.
+
+
+ */
+
+insert into system_metrics (trial_id, end_time, metrics, trial_run_id, partition_type, metric_group)
+-- Long-winded query to parse and massage existing data in `trial_profiler_metrics` to fit the new schema
+select
+    pm.trial_id as trial_id,
+    -- Use latest time of aggregate GPU groups
+    max(pm.ts) as end_time,
+    (
+        case pm.mgroup
+            when 'gpu' then jsonb_build_object(
+                    pm.agent_id,
+                    jsonb_object_agg(
+                        -- GPU UUID should never be null here, but SQL doesn't know that.
+                            coalesce(pm.gpu_uuid, ''), pm.metrics
+                        )
+                )
```

Comment:
indentation here is a bit weird. I'd expect

```suggestion
            when 'gpu' then jsonb_build_object(
                pm.agent_id,
                jsonb_object_agg(
                    -- GPU UUID should never be null here, but SQL doesn't know that.
                    coalesce(pm.gpu_uuid, ''), pm.metrics
                )
            )
```


I remember Hamid added an SQL formatter: https://github.com/determined-ai/determined/pull/7538/files
maybe run it and see how it comes out?

Reply:
i ran that and it formatted just about everything _except_ this SQL file... in any case, i just formatted this by hand.